src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1512,6 +1532,358 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
+                                              (base64.b64encode(bytes("?" + sas_token, 'ascii'))).decode('ascii'))
+    container_logs = "azure-arc"
+    kube_objects = "azure-arc/pod azure-arc/service azure-arc/deployment"
+    yaml_lines = deployment_yaml.splitlines()
+    for index, line in enumerate(yaml_lines):
+        if "DIAGNOSTIC_CONTAINERLOGS_LIST" in line and container_logs is not None:
+            yaml_lines[index] = line + ' ' + container_logs
+        if "DIAGNOSTIC_KUBEOBJECTS_LIST" in line and kube_objects is not None:
+            yaml_lines[index] = line + ' ' + kube_objects
+        if "CLUSTER_TYPE" in line:
+            yaml_lines[index] = '  CLUSTER_TYPE: connectedCluster'
+
+    deployment_yaml = '\n'.join(yaml_lines)
+
+    fd, temp_yaml_path = tempfile.mkstemp()
+    temp_yaml_file = os.fdopen(fd, 'w+t')
+    try:
+        temp_yaml_file.write(deployment_yaml)
+        temp_yaml_file.flush()
+        temp_yaml_file.close()
+        print(temp_yaml_file)
+        try:
+            kube_config = None
+            print("Cleaning up aks-periscope resources if existing")
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "serviceaccount,configmap,daemonset,secret",
+                             "--all", "-n", "aks-periscope", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+            
+            subprocess_cmd = ["kubectl", "delete",
+            "ClusterRoleBinding",
+            "aks-periscope-role-binding", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "ClusterRole",
+                             "aks-periscope-role", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+        
+            subprocess_cmd = ["kubectl", "delete",
+                             "--all",
+                             "apd", "-n", "aks-periscope", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            subprocess_cmd = ["kubectl", "delete",
+                             "CustomResourceDefinition",
+                             "diagnostics.aks-periscope.azure.github.com", "--ignore-not-found"]
+            if kube_config:
+                subprocess_cmd.extend(["--kubeconfig", kube_config])
+            subprocess.call(subprocess_cmd, stderr=subprocess.STDOUT)
+
+            print()
+            print("Deploying aks-periscope")
```

Comment:
Try to avoid printing "aks-periscope" to the end user, since the cluster being onboarded will not always be an AKS cluster. Use something generic like "Deploying diagnostic container on the K8s cluster".